ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/ClusterStatusResource.java

```diff
@@ -119,6 +122,19 @@ private HostStoreLags getHostStoreLags(final KsqlHostInfo ksqlHostInfo, final bo
     return perQueryMap;
   }
 
+  private Map<String, Set<TaskMetadata>> getTaskMetadata(
+          final KsqlHostInfo ksqlHostInfo
+  ) {
+    final Map<String, Set<TaskMetadata>> perQueryMap = new HashMap<>();
+    for (PersistentQueryMetadata persistentQueryMetadata: engine.getPersistentQueries()) {
```

Comment:
This was the best way I found to get to the `PersistentQueryMetadata` to call `getTaskMetadata()` please let me know if there is a better way

Reply:
`engine` is the right place to get the persistent queries. Btw, this is only getting the queries from the current ksql node. What about the rest of the nodes? If the query is idle in this node but active in another, this won't return the uptime of the active one.

Reply:
That is good. I think the idea was to only check the active queries and then we aggregate them in the health check and send them to dd and metabase